File: fixtures/js/dead_store/negative.ts
// MAINT012-dead-store negative fixture

function readAfterWrite() {
    let x = 1;
    return x;   // x is read — not dead
}

function underscorePrefix() {
    let _x = 1;
    return null;   // skipped — leading underscore
}

function loopVar(arr: number[]) {
    for (let x of arr) {
        // loop var — skipped
    }
}

function destructure(obj: { a: number; b: number }) {
    const { a, b } = obj;
    return a + b;   // both a and b are used
}

File: fixtures/rust/dead_store/negative.rs
// MAINT012-dead-store negative fixture

fn read_after_write() {
    let x = 1;
    let _ = x + 1;  // x is read — not dead
}

fn underscore_prefix() {
    let _x = 1;   // leading underscore — skipped
}

fn mut_binding() {
    let mut x = 1;
    x = 2;
    let _ = x;  // mut skipped
}

fn destructure() {
    let pair = (1, 2);
    let _ = pair;
}
